state-synchronizer/src/executor_proxy.rs

```diff
@@ -47,19 +50,38 @@ pub trait ExecutorProxyTrait: Sync + Send {
 
     /// Tries to find a LedgerInfo for a given version.
     async fn get_ledger_info(&self, version: u64) -> Result<LedgerInfoWithSignatures>;
+
+    /// publishes on-chain config updates to subscribed components
+    async fn publish_on_chain_config_updates(&mut self, events: Vec<ContractEvent>);
 }
 
 pub(crate) struct ExecutorProxy {
     storage_read_client: Arc<StorageReadServiceClient>,
     executor: Arc<Executor<LibraVM>>,
+    reconfig_subscriptions: Vec<Box<dyn EventSubscription>>,
+    subscribed_configs_union: HashMap<ConfigID, Vec<u8>>,
```

Comment:
nit: maybe let's just call it `on_chain_config`. Cause it basically all on-chain config values known to node software